### PR TITLE
Remove session caching and state

### PR DESCRIPTION
**Pull request by weikanglim**
Removes `container` and `manifest` caching from the session. With this change, each RPC request creates a container copy with only registrations and recreates components fully to handle the particular request. This sidesteps concurrency issues with component designs that may not be handled fully.

This effectively means all components are "request scoped" and would perform the same app initialization logic as-if the "azd cli" process had started from brand new. This allows for concurrency and avoids blocking, at the cost of memory + I/O redundancy. This is an expedient change to increase reliability and avoid any caching issues. In the future, as we start refactoring the app to separate out different systems that are initialized differently, we can reshare components + implement different I/O polling strategies for the complete e2e to work more efficiently.

Addresses #3287 , #3297, #3362

### CODE REVIEW

**Comment by rajeshkamal5050** [comment]
Wei, Will it also address #3358 ?

**Comment by weikanglim** [comment]
@rajeshkamal5050 Yes, indeed. Closed that as duplicate of what is mentioned in #3297 "RefreshEnvironmentAsync returns deployment status for a different environment than the one requested"

**Review by ellismg — APPROVED** [review]
One small comment about hard coding the playground subscription ID in the end to end test.

Excited to see this land and see how much worse performance becomes.  I think with the change in our APIs to give VS ways to control how much data is fetched we won't have a problem here - thanks for moving us back to a known good state and challenging the use of caching when we weren't sure we needed it.

File: cli/azd/test/functional/vs_server_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			err = json.Unmarshal(stdout.Bytes(), &svr)
+			require.NoError(t, err, "value: %s", stdout.String())
+
+			/* #nosec G204 - Subprocess launched with a potential tainted input or cmd arguments false positive */
+			cmd = exec.CommandContext(context.Background(),
+				"dotnet", "test",
+				"--filter", "Name="+tt)
+			cmd.Dir = testDir
+			cmd.Env = append(cmd.Env, os.Environ()...)
+			cmd.Env = append(cmd.Env, "AZURE_SUBSCRIPTION_ID="+"faa080af-c1d8-40ad-9cce-e1a450ca5b57")
```
[the remark this answers is not in the document]
Comment (weikanglim) [reply cli/azd/test/functional/vs_server_test.go:88]: Oops! A ToDo that should be removed. One unfortunate part is that we generally hinge on the default subscription from `azd` for these cases. I might implement that.

**Review by vhvb1989 — APPROVED** [review]
LGTM

**Pull request merged**